### ISSUE TITLE
Does WDRBD 1.32 support the connection of WDRBD and Linux DRBD？

### ISSUE THREAD

**Issue by sunxingwen4567** [mantechnology/wdrbd#17]
Does WDRBD 1.32 support the connection of WDRBD and Linux DRBD？

Thanks

**Comment by sungeun-kim** [comment]
Sorry, Not yet.

**Comment by mailinglists35** [comment]
@sungeun-kim if I wanted to be among the first to know when it will work with linux, subscribing to this issue should be enough, or will I have to sign-up somewhere else? (watch the master commits? some other easy way to get notified about this?)

thank you

**Comment by sungeun-kim** [comment]
Hi
maybe next month is starting point to solve this issue. 
we are very busy to do something else.

first, I register this issue to github projects page. you will see to solve this.
https://github.com/mantechnology/wdrbd/projects/1 

Could I ask a question ? why do you want to be this function?

thanks.

**Comment by sunxingwen4567** [comment]
Hi
WDRBD and LDRBD may connect with each other successfully. I can do this via changing 2 pieces of code of WDRBD one week ago. And, WDRBD and LDRBD may copy the volume data to each other, however, the action of copy may corrupt the volume, and make the volume unable to be open again.

The reason for this function is that: WINDOWS may serve as a Customer, while Linux serves as a Server.  It is required that the modified data on some Windows volume may be quickly copied and saved into one Linux volume, by DRBD.

**Comment by sungeun-kim** [comment]
which filesystem do you use on linux side and Windows side? or raw filesystem?

**Comment by sunxingwen4567** [comment]
NTFS is used on Windows 7, while the default filesystem(mkfs.ext4 /dev/drbd0), is used on CentOS Linux.

Fortunately, I found that now, executing the command of "drbdadm primary --force r0" on LDRBD on CentOS, may make it successful to recover(copy) the data from /dev/drbd0 on CentOS back to F partition on  Windows 7. The data on /dev/drbd0 on CentOS are originally copied from F partition on Windows 7. And the data recovered back onto the F partition of Windows 7, can work.

**Comment by sunxingwen4567** [comment]
Another question:

DW-1142: don't start 'invalidate' if peer is not primary  

Why ?

The function of "invalidate if both local Windows and peer Linux are secondary" is exactly what I want at present.

**Comment by sungeun-kim** [comment]
WDRBD's invalidate is required to be peer's primary. we intentionally modified it for a some reason. It's complicated to explain.

**Comment by mailinglists35** [comment]
> Could I ask a question ? why do you want to be this function?

@sungeun-kim I also want to do something similar:

windows ntfs on top of a drbd volume.
the drbd volume will be made of:
- one leg physical disk connected to windows machine
- one leg a zfs zvol in a linux machine

the linux will be secondary and the windows will be primary

then I will be able to live snapshot the linux zfs leg so I can use the zfs volume as an instant incremental image snapshot for the ntfs volume. and in case of recovery i could readonly mount an earlier zfs snapshot inside linux using ntfs-3g and restore the data via samba to the windows machine.

so basically this means cheap near-continuous data protection for windows machines.

**Comment by mailinglists35** [comment]
@sunxingwen4567 can you post your modifications of code? it would be awesome. thank you!

ps: if you want windows to not touch a disk, go to diskpart and set it offline. this way you can use it in third party apps (iscsi target, virtualization, wdrbd etc) and never attempt to be mounted locally.

**Comment by mailinglists35** [comment]
@sungeun-kim please don't close this. having primary node on windows and secondary node on linux is the whole reason of hoping for a working windows drbd.
it's very dissapointing if you will never allow this.
it's like the whole effort you put into porting this for windows is in vain. a very big big dissapointment!
:(

**Comment by sungeun-kim** [comment]
@mailinglists35 you don't need to disappoint this (just closing issue).
we have a plan to support this issue.  will reopen this at that time.

**Comment by mailinglists35** [comment]
@sungeun-kim thank you for the clarification! looking forward for that moment!

**Comment by WonSunMantech** [comment]
Add a setting to the resource that increase the ping-timeout for the latest drbd and the latest windows drbd interworking. 
but actual operation will not be due to logic differences.

**Comment by mailinglists35** [comment]
> sungeun-kim reopened this a day ago

whoa thank you!
